src/app/digital/actions/ports/CoderPortChangeAction.ts

```diff
@@ -19,25 +25,40 @@ export class CoderPortChangeAction implements Action {
     protected inputPortAction: InputPortChangeAction;
     protected outputPortAction: OutputPortChangeAction;
 
+ /**
+  * The constrcutor determines whether the object is an encoder or decoder and calls on the corresponding actions based on the type.
+  *
+  * @param obj is the encoder or decoder selected
+  * @param initial the inital number of outputs/inputs
+  * @param target the target numbe of outputs/inputs
+  */
     public constructor(obj: Encoder | Decoder, initial: number, target: number) {
         this.obj = obj;
         this.targetCount = target;
 
         if (obj instanceof Encoder) {
             this.initialCount = obj.getOutputPortCount().getValue();
-            this.inputPortAction  = new InputPortChangeAction(obj, obj.getInputPortCount().getValue(), Math.pow(2, target));
+            this.inputPortAction  = new InputPortChangeAction(obj, initial, Math.pow(2, target));
             this.outputPortAction = new OutputPortChangeAction(obj, initial, target);
         } else {
             this.initialCount = obj.getInputPortCount().getValue();
             this.inputPortAction  = new InputPortChangeAction(obj, initial, target);
-            this.outputPortAction = new OutputPortChangeAction(obj, obj.getOutputPortCount().getValue(), Math.pow(2, target));
+            this.outputPortAction = new OutputPortChangeAction(obj, initial, Math.pow(2, target));
```

Comment:
Were the first and last lines in this selection also supposed to be changed?

Reply:
I don't think so, I'm will change it to the original since I was only supposed to do documentation